docs/cep/tutorials/consuming-data.md

```diff
@@ -180,8 +183,9 @@ Stream processor consumes a message in the default format when it makes no chang
 
 Stream processor consumes a message in the custom format when it makes changes to the names of the attributes of the message schema before it processes the message. To understand how messages are consumed in custom format, follow the procedure below.
```

Comment:
Stream processor, part of stream worker used for query part, we have sample documentation here: https://macrometa.com/docs/cep/query-guide/functions/rdbms/query , also I would leave it as it is

Reply:
But that is specifically for the RDBMS plugin. Is this topic about using that plugin?

Reply:
_So this should be a proper explanation of stream processors:_
Stream processors are a combination of stream functions and windows. They work directly on the input streams (or named-windows), to generate zero or more new events with zero or more additional output attributes while having the ability to retain and arbitrarily emit events.
They are more advanced than stream functions as they can retain and arbitrarily emit events, and they are more advanced than windows because they can add additional attributes to the events.
_Purpose:_
Stream processors help to achieve complex execution logics that cannot be achieved by other constructs such as functions, aggregate functions, stream functions, and windows.

Reply:
sure